pkg/scheduler/framework/session_plugins.go

```diff
@@ -638,14 +637,59 @@ func (ssn *Session) PredicateFn(task *api.TaskInfo, node *api.NodeInfo) ([]*api.
 			if !found {
 				continue
 			}
-			status, err := pfn(task, node)
-			predicateStatus = append(predicateStatus, status...)
+			err := pfn(task, node)
 			if err != nil {
-				return predicateStatus, err
+				return err
 			}
 		}
 	}
-	return predicateStatus, nil
+	return nil
+}
+
+// UnscheduleAndUnresolvedError checks if the predicate error contains
+// - Unschedulable
+// - UnschedulableAndUnresolvable
+// - ErrorSkipOrWait
+func (ssn *Session) UnscheduleAndUnresolvedError(task *api.TaskInfo, node *api.NodeInfo) error {
```

Comment:
Put these to file `session.go` is better.

Reply:
Done.